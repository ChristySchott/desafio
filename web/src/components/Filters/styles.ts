import styled from 'styled-components';

export const Container = styled.div`
  width: 280px;
  margin: 10px auto;
`;

export const Page = styled.div`
  margin-bottom: 10px;

  a {
    color: #100c14;
  }

  span {
    color: #c61a28;
  }
`;

export const Menu = styled.div`
  width: 100%;
  padding: 15px;
  border: 1px solid #ddd;
  border-radius: 2px;

  h2,
  h3 {
    text-transform: uppercase;
  }

  h2 {
    color: #c61a28;
    line-height: 1;
  }

  h3 {
    color: #04acac;
    margin-bottom: 12px;
  }

  ul {
    margin-left: 20px;

    li {
      padding: 0;
    }

    a {
      color: #888;
    }

    li + li {
      margin-top: 5px;
    }
  }

  h3:not(:first-child) {
    margin-top: 16px;
  }
`;

export const Colors = styled.ul`
  display: flex;
  list-style: none;
  margin: 0 !important;

  li {
    width: 50px;
    height: 20px;
    background: #fa4115;
    margin: 0 !important;
  }

  li:nth-child(2) {
    background: #c61a28;
  }

  li:nth-child(3) {
    background: #04acac;
  }

  li + li {
    margin-left: 5px !important;
  }
`;
