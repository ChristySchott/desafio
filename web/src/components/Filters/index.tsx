/* eslint-disable jsx-a11y/control-has-associated-label */
/* eslint-disable jsx-a11y/anchor-has-content */
import React from 'react';

import { useFilter } from 'hooks/filter';
import { useColorFilter } from 'hooks/color';
import { Container, Menu, Colors } from './styles';

const Filters: React.FC = () => {
  const { setFilter } = useFilter();
  const { setColorToFilter } = useColorFilter();

  return (
    <Container>
      <Menu>
        <h2>Filtre por</h2>

        <h3>Categorias</h3>
        <ul>
          <li>
            <a href="/">Roupas</a>
          </li>
          <li>
            <a href="/">Sapatos</a>
          </li>
          <li>
            <a href="/">Acess√≥rios</a>
          </li>
        </ul>

        <h3>Cores</h3>
        <Colors>
          <li>
            <button
              type="button"
              onClick={() => setColorToFilter('Vermelho')}
            />
          </li>
          <li>
            <button type="button" onClick={() => setColorToFilter('Amarelo')} />
          </li>
          <li>
            <button type="button" onClick={() => setColorToFilter('Azul')} />
          </li>
        </Colors>

        <h3>Tipo</h3>
        <ul>
          <li>
            <button type="button" onClick={() => setFilter('Corrida')}>
              Corrida
            </button>
          </li>
          <li>
            <button type="button" onClick={() => setFilter('Caminhada')}>
              Caminhada
            </button>
          </li>
          <li>
            <button type="button" onClick={() => setFilter('Casual')}>
              Casual
            </button>
          </li>
          <li>
            <button type="button" onClick={() => setFilter('Social')}>
              Social
            </button>
          </li>
        </ul>
      </Menu>
    </Container>
  );
};

export default Filters;
