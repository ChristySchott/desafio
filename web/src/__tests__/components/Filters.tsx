import React from 'react';
import { render } from '@testing-library/react';

import Filters from '../../components/Filters';

describe('<Filters />', () => {
  it('should render with no error', () => {
    const { container } = render(<Filters />);

    expect(container).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
