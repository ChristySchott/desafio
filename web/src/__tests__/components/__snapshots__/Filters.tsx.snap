// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> should render with no error 1`] = `
<div
  class="sc-bdnylx bcOrZQ"
>
  <div
    class="sc-gtssRu jQOmBW"
  >
    <h2>
      Filtre por
    </h2>
    <h3>
      Categorias
    </h3>
    <ul>
      <li>
        <a
          href="/"
        >
          Roupas
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Sapatos
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Acess√≥rios
        </a>
      </li>
    </ul>
    <h3>
      Cores
    </h3>
    <ul
      class="sc-dlnjPT iRkRgz"
    >
      <li>
        <a
          href="/"
        />
      </li>
      <li>
        <a
          href="/"
        />
      </li>
      <li>
        <a
          href="/"
        />
      </li>
    </ul>
    <h3>
      Tipo
    </h3>
    <ul>
      <li>
        <a
          href="/"
        >
          Corrida
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Caminhada
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Casual
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Social
        </a>
      </li>
    </ul>
  </div>
</div>
`;
